aws/rust-runtime/aws-config/src/provider_config.rs

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+//! Configuration Options for Credential Providers
+
+use aws_hyper::DynConnector;
+use aws_types::os_shim_internal::{Env, Fs};
+use aws_types::region::Region;
+use smithy_async::rt::sleep::AsyncSleep;
+use std::sync::Arc;
+
+/// Configuration options for Credential Providers
+///
+/// Most credential providers builders offer a `configure` method which applies general provider configuration
+/// options.
+///
+/// To use a region from the default region provider chain use [`ProviderConfig::with_default_region`].
+/// Otherwise, use [`ProviderConfig::without_region`]. Note that some credentials providers require a region
+/// to be explicitly set.
+#[derive(Clone, Default)]
+pub struct ProviderConfig {
+    env: Env,
+    fs: Fs,
+    connector: Option<DynConnector>,
+    sleep: Option<Arc<dyn AsyncSleep>>,
+    region: Option<Region>,
+}
+
+impl ProviderConfig {
+    /// Create a default provider config with the region unset
+    ///
+    /// Using this option means that you may need to set a region manually.
+    ///
+    /// # Example
+    /// ```rust
+    /// use aws_config::provider_config::ProviderConfig;
+    /// use aws_sdk_sts::Region;
+    /// use aws_config::web_identity_token::WebIdentityTokenCredentialsProvider;
+    /// let conf = ProviderConfig::without_region().with_region(Some(Region::new("us-east-1")));
```

Comment:
`without_region` followed by `with_region` is a bit of a head scratcher, but I'm not sure I have a better suggestion. Maybe `without_default_region`, but then you can follow that with `load_default_region` and have the same problem.